File: java/src/com/thoughtworks/cep/accounts/CheckingAccount.java
package com.thoughtworks.cep.accounts;

public class CheckingAccount {
  public CheckingAccount(long amount) {}

  long balance() {
    throw new UnsupportedOperationException();
  }

  void withdraw(int amount) {
    throw new UnsupportedOperationException();
  }

  void deposit(int amount) {
    throw new UnsupportedOperationException();
  }
}

File: kotlin/src/com/thoughtworks/cep/accounts/CheckingAccount.kt
package com.thoughtworks.cep.accounts

class CheckingAccount(openingBalance: Long = 0) {
    fun balance(): Long {
        throw UnsupportedOperationException()
    }

    fun withdraw(amount: Int) {
        throw UnsupportedOperationException()
    }

    fun deposit(amount: Int) {
        throw UnsupportedOperationException()
    }
}